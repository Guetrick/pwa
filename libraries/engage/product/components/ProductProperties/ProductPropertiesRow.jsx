import React from 'react';
import PropTypes from 'prop-types';
import { tableCell } from './ProductProperties.style';

/**
 * @param {Object} props The component props.
 * @return {JSX}
 */
export function ProductPropertiesRow({ label, value }) {
  return (
    <tr key={`${label}${value}`}>
      <td className={tableCell}>{label}</td>
      <td className={tableCell} data-test-id={`property: ${value}`}>{value}</td>
    </tr>
  );
}

ProductPropertiesRow.propTypes = {
  label: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
};
