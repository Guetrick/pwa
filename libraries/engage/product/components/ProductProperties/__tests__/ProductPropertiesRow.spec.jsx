import React from 'react';
import { shallow } from 'enzyme';
import { ProductPropertiesRow } from '../ProductPropertiesRow';

describe('<ProductPropertiesRow />', () => {
  it('should render as expected', () => {
    const wrapper = shallow(<ProductPropertiesRow label="TestLabel" value="TestValue" />);
    expect(wrapper.find('td').at(0).text()).toEqual('TestLabel');
    expect(wrapper.find('td').at(1).text()).toEqual('TestValue');
    expect(wrapper).toMatchSnapshot();
  });
});
