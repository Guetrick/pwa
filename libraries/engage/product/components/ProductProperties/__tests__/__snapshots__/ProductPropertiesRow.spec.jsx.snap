// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductPropertiesRow /> should render as expected 1`] = `
<tr
  key="TestLabelTestValue"
>
  <td
    className={
      Object {
        "data-css-fl9ufm": "",
      }
    }
  >
    TestLabel
  </td>
  <td
    className={
      Object {
        "data-css-fl9ufm": "",
      }
    }
    data-test-id="property: TestValue"
  >
    TestValue
  </td>
</tr>
`;
